mne/cov.py

```diff
@@ -1656,32 +1691,13 @@ def whiten_evoked(evoked, noise_cov, picks=None, diag=False, rank=None,
     evoked = evoked.copy()
     if picks is None:
         picks = pick_types(evoked.info, meg=True, eeg=True)
-    W = _get_whitener_data(evoked.info, noise_cov, picks,
-                           diag=diag, rank=rank, scalings=scalings)[0]
-    evoked.data[picks] = np.sqrt(evoked.nave) * np.dot(W, evoked.data[picks])
-    return evoked
-
 
-@verbose
-def _get_whitener_data(info, noise_cov, picks, diag=False, rank=None,
-                       scalings=None, verbose=None):
-    """Get whitening matrix for a set of data."""
-    ch_names = [info['ch_names'][k] for k in picks]
-    # XXX this relies on pick_channels, which does not respect order,
-    # so this could create problems if users have reordered their data
-    noise_cov = pick_channels_cov(noise_cov, include=ch_names, exclude=[])
-    if len(noise_cov['data']) != len(ch_names):
-        missing = list(set(ch_names) - set(noise_cov['names']))
-        raise RuntimeError('Not all channels present in noise covariance:\n%s'
-                           % missing)
-    if diag:
-        noise_cov = noise_cov.copy()
-        noise_cov['data'] = np.diag(np.diag(noise_cov['data']))
+    scalings = _handle_default('scalings', None)
```

Comment:
It looks like this code path (for `whiten_evoked`) previously used:
```
_get_whitener_data
```
Which in turn used:
```
scalings = _handle_default('scalings_cov_rank', scalings)
```
but now you have it just do:
```
scalings = _handle_default('scalings', None)
```
Shouldn't this still be `scalings = _handle_default('scalings_cov_rank', scalings)` here?

If so, I don't think you even need it because `compute_whitener` does this default check internally on line 1641.

Reply:
good catch